### PR TITLE
Teach the user to write the command lazy loading

### PR DESCRIPTION
**Pull request by SimonHeimberg**
A lazy loaded command is instantiated only when it is called.

### CODE REVIEW

**Comment by SimonHeimberg** [comment]
about lazy loading a command: https://symfony.com/doc/current/console/commands_as_services.html#lazy-loading

**Comment by ndench** [comment]
Are lazy loaded commands better in every situation? Should this contain a link to the `commands_as_services` docs?

File: console.rst [hunk 1]
```diff
@@ -56,10 +59,6 @@ method. Then you can optionally define a help message and the
     // ...
     protected function configure()
     {
-        $this
```
[the remark this answers is not in the document]
Comment (noniagriconomie) [reply console.rst]: yes, ->setName(self::$defaultName) is needed

[the remark this answers is not in the document]
Comment (apfelbox) [reply console.rst]: You don‘t need the setName call if you have the static property, but you still need $this

[the remark this answers is not in the document]
Comment (SimonHeimberg) [reply console.rst]: done, thanks

**Comment by SimonHeimberg** [comment]
A link to https://symfony.com/doc/current/console/commands_as_services.html#lazy-loading sounds good, @ndench. Where should its place be?

I do not know if lazy loading is a good idea in all situations. As mentioned in `commands_as_services`, `->addAlias()` must be handled. Any suggestions of situations?

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
I think I would not add the link. This is just working and adding a cross-reference to another article may be more confusing.

**Comment by ndench** [comment]
I'm not fussed either way. As long as there is somewhere in the docs that explain that lazy loading isn't the only way to write commands.

**Comment by kunicmarko20** [comment]
Maybe instead of replacing the non lazy way, add another example under it where the lazy load is mentioned with a static property?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
This should really be the way to go 99% of the time

**Comment by xabbuh** [comment]
Thank you @SimonHeimberg.

**Pull request closed without merging**